crates/sncast/src/starknet_commands/account/import.rs

```diff
@@ -98,36 +98,62 @@ pub async fn add(
         _ => fetched_class_hash,
     };
 
-    let legacy = check_if_legacy_contract(class_hash, add.address, provider).await?;
+    let chain_id = get_chain_id(provider).await?;
+    if import.salt.is_some() && class_hash.is_some() {
+        let sncast_account_type = match import.account_type {
+            AccountType::Argent => SNCastAccountType::Argent,
+            AccountType::Braavos => SNCastAccountType::Braavos,
+            AccountType::Oz => SNCastAccountType::OpenZeppelin,
+        };
```

Comment:
note: As described in https://github.com/foundry-rs/starknet-foundry/issues/2571, there are two `AccountType` enums, unfortunately we need to map one to another here. This should be also refactored in mentioned issue.

Reply:
Let's add relevant TODO to the code with issue number